# 误差控制语句-尝试，捕捉

> 原文:[https://www . javatpoint . com/MATLAB-error-control-statement-try-catch](https://www.javatpoint.com/matlab-error-control-statement-try-catch)

MATLAB 定义了一些用来控制误差的函数。try-catch 语句是一个错误控制函数，如下所述。

## Try - catch 语句

Try-catch 语句提供错误处理控制。try-catch 语句的一般形式是

**语法:**

```

	try
		Statements
	catch   exception
		Statements
	end

```

try 和 catch 之间的语句首先执行。如果在 try 和 catch 之间执行语句时没有出现错误，则 MATLAB 会进一步执行 end 关键字之后的语句/代码。如果在 try 和 catch 之间的语句执行过程中出现错误，则 MATLAB 会在 catch 和 end 之间执行语句。Try-catch 语句可以通过下面的例子来解释。

**示例:**

```

a = ones(4);
b = zeros(3);
try
    c = [a;b];
catch ME
    disp(ME)
end

```

**输出:**

```

 MException with properties:

    identifier: 'MATLAB:catenate:dimensionMismatch'
       message: 'Dimensions of arrays being concatenated are not consistent.'
         cause: {01 cell}
         stack: [31 struct]
    Correction: []

```

![MATLAB Error Control Statement-try, catch](../Images/446393b2370e5a2ec2966334dcc0e277.png)

**以下是在 MATLAB 中使用 try/catch 语句时的要点:**

*   程序执行的控制首先进入 try 块并执行每个语句。
*   如果在执行过程中出现任何错误，则控制会立即传递给 catch 块，而 try 块的任何其他语句都不会执行。
*   如果 try 块内没有出现错误，则控件不会进入 catch 块。然后，控件到达 try/ catch 块的 end 关键字之后的语句。
*   每当 try 块中出现任何错误或异常，MATLAB 都会构造一个**meexception**类的实例，并在 catch 语句中返回该对象。
*   可以用变量 **ME** 来访问 MException 类对象。
*   MException 类对象有五个属性- **标识符、消息、堆栈、原因、**和**修正**。这些属性描述了发生的异常的详细信息。
*   我们不能使用多个 catch 块，一个 try 块中只允许有一个 catch 块。
*   如果需要，我们可以使用嵌套的 try/catch 块。

**显示 MException 类对象属性的示例:**

```

a = ones(4);
b = zeros(3);
try
    c = [a;b];
catch ME
   % strcmp string compare function, to check the same string in ME identifier
    if (strcmp(ME.identifier,'MATLAB:catenate:dimensionMismatch'))
	% if it is true then change the message property
        msg = ['dimension mismatch occured: First argument has ',...
            num2str(size(a,2)), ' columns, while second argument has ',...
            num2str(size(b,2)),' columns.'];
        causeException = MException('MATLAB:mycode:dimensions',msg)
        ME = addCause(ME, causeException);
    end

end

```

**输出:**

```
causeException = 
  MException with properties:

    identifier: 'MATLAB:mycode:dimensions'
       message: 'dimension mismatch occured: First argument has 4 columns, while second argument has 3 columns.'
         cause: {}
         stack: [01 struct]
    Correction: []

```

## 程序终止

程序终止控制允许在正常终止点之前的某个点退出我们的程序。

* * *